graphein/datasets/pdb_data.py

```diff
@@ -0,0 +1,1092 @@
+            first_df_split, second_df_split, how="inner", on=[df_primary_key]
+        )
+        return merged_df_split
+
+    def split_clusters(
+        self,
+        df: pd.DataFrame,
+        update: bool = False,
+    ) -> Dict[str, pd.DataFrame]:
+        """Split clusters derived by MMseqs2.
+
+        :param df: DataFrame containing the clusters derived by MMseqs2.
+        :type df: pd.DataFrame
+        :param update: Whether to update the selection to the representative
+            sequences, defaults to ``False``.
+        :type update: bool, optional
+
+        :return: A Dictionary of split names mapping to DataFrames of
+            randomly-split representative sequences.
+        :rtype: Dict[str, pd.DataFrame]
+        """
+        split_ratios_provided = self.split_ratios is not None
+        assert split_ratios_provided
+
+        # Split clusters
+        log.info(
+            f"Randomly splitting clusters into ratios: {' '.join([str(r) for r in self.split_ratios])}..."
+        )
+        df_splits = self.split_df_proportionally(
+            df,
+            self.splits,
+            self.split_ratios,
+            self.assign_leftover_rows_to_split_n,
+        )
+        log.info("Done splitting clusters")
+
+        # Update splits
+        for split in self.splits:
+            if update:
+                df_split = df_splits[split]
+                if self.df_splits[split] is not None:
+                    self.df_splits[split] = self.merge_df_splits(
+                        self.df_splits[split], df_split
+                    )
+                else:
+                    self.df_splits[split] = df_split
+                df_splits[split] = self.df_splits[split]
+
+        return df_splits
+
+    def cluster(
+        self,
+        min_seq_id: float = 0.3,
+        coverage: float = 0.8,
+        update: bool = False,
+    ) -> Union[pd.DataFrame, Dict[str, pd.DataFrame]]:
+        """Cluster sequences in selection using MMseqs2.
+
+        :param min_seq_id: Sequence identity, defaults to ``0.3``.
+        :type min_seq_id: float, optional
+        :param coverage: Clustering coverage, defaults to ``0.8``.
+        :type coverage: float, optional
+        :param update: Whether to update the selection to the representative
+            sequences, defaults to ``False``.
+        :type update: bool, optional
+        :return: Either a single DataFrame of representative sequences or a
+            Dictionary of split names mapping to DataFrames of randomly-split
+            representative sequences.
+        :rtype: Union[pd.DataFrame, Dict[str, pd.DataFrame]]
+        """
+        # Write fasta
+        self.to_fasta("pdb.fasta")
+        if not is_tool("mmseqs"):
+            log.error(
+                "MMseqs2 not found. Please install it: conda install -c conda-forge -c bioconda mmseqs2"
+            )
+
+        # Create clusters
+        if not os.path.exists("pdb_cluster_rep_seq.fasta"):
```

Comment:
Hmm, this can cause problems with multiple runs.

I think a starting solution is to include the clustering params in the FASTA file (e.g.` pdb_cluster_rep_seq_id_0.3_c_0.8.fasta`). As well as log statements to make it clear whether a file is being read or created.

Reply:
Agreed. I will most likely work on other feature requests in the meantime, so as not duplicate work, may I ask for you to integrate this when you are available to do so?